Sources/Commands/SwiftScriptTool.swift

```diff
@@ -0,0 +1,374 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import ArgumentParser
+import Basics
+import PackageModel
+import ScriptParse
+import ScriptingCore
+import TSCBasic
+import Workspace
+
+protocol ScriptCommand: ParsableCommand {
+    var swiftOptions: SwiftToolOptions { get }
+    var options: ScriptToolOptions { get }
+
+    func run(_ swiftTool: SwiftTool, as productName: String, at cacheDirPath: AbsolutePath) throws
+}
+
+extension ScriptCommand {
+    public func run() throws {
+        guard let file = options.file else {
+            throw ScriptError.fileNotFound("")
```

Comment:
Should this perhaps be a different error, indicating that no file was provided (as opposed to that the provided file couldn't be found)?